### PR TITLE
refactor comments so post reply is always open

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

part of #125 

#### What's this PR do?

In the mocks the 'reply' textfield for a top-level comment (i.e. a comment replying to a post) is supposed to be always open. Currently, it's not, and you need to click a 'reply' button to open it and start composing a comment.

This refactors the code for rendering the commenting form, and separates it into two separate components, one for comment replies and one for post replies. 

#### How should this be manually tested?

Commenting on posts, replying to comments etc should all work as expected.

### CODE REVIEW

File: static/js/components/CreateCommentForm_test.js [hunk 1]
```diff
@@ -49,124 +56,221 @@ describe("CreateCommentForm", () => {
...
-            helper
-              .listenForActions([forms.FORM_UPDATE], () => {
-                wrapper.find("textarea[name='text']").simulate("change", {
-                  target: {
-                    name:  "text",
-                    value: expectedKeys.text
-                  }
-                })
+    it("should cancel and hide the form", () => {
+      return helper
```
Comment (rhysyngsun) [inline static/js/components/CreateCommentForm_test.js]: Should this and the following tests get refactored to the `async`/`await` styles while you're touching them here?

File: static/js/components/CreateCommentForm.js [hunk 2]
```diff
@@ -15,123 +15,199 @@ type CreateCommentFormProps = {
...
+    : replyToPostKey(ownProps.post)
+
+const mapStateToProps = (state, ownProps) => {
+  const initialValue = ownProps.comment
+    ? getCommentReplyInitialValue(ownProps.comment)
+    : getPostReplyInitialValue(ownProps.post)
+
+  return {
+    post:    ownProps.post || state.posts.data.get(initialValue.post_id),
+    initialValue,
```
[the remark this answers is not in the document]
Comment (aliceriot) [reply static/js/components/CreateCommentForm.js]: ahaha, will do

File: static/js/components/CreateCommentForm_test.js (hunk 1 above)
Comment (aliceriot) [reply static/js/components/CreateCommentForm_test.js]: probably not a bad idea, I'll do that now

Comment (aliceriot) [reply static/js/components/CreateCommentForm_test.js]: I really prefer the `async / await` sugar, just avoiding all those `.then` nested callbacks makes the code so much more readable IMO

**Comment by rhysyngsun** [comment]
Functionality looks good

**Comment by aliceriot** [comment]
comments addressed!

File: static/js/components/CreateCommentForm_test.js (hunk 1 above)
Comment (rhysyngsun) [reply static/js/components/CreateCommentForm_test.js]: agreed

**Pull request merged**